app/controllers/api/services_controller.rb

```diff
@@ -19,6 +27,61 @@ def reconfigure_resource(type, id = nil, data = nil)
 
     private
 
+    def build_service_attributes(data)
+      attributes                           = data.dup
+      attributes['orchestration_manager']  = fetch_ext_management_system(data['orchestration_manager']) if data['orchestration_manager']
+      attributes['orchestration_template'] = fetch_orchestration_template(data['orchestration_template']) if data['orchestration_template']
+      attributes['job_template']           = fetch_configuration_script(data['job_template']) if data['job_template']
+      attributes['parent']                 = fetch_service(data['parent_service']) if data['parent_service']
+      if data['job_options']
+        # AnsibleTowerClient needs the keys to be symbols
+        attributes['job_options'][:limit]      ||= data['job_options'].delete('limit')
+        attributes['job_options'][:extra_vars] ||= data['job_options'].delete('extra_vars')
+      end
+      attributes.delete('parent_service')
+      attributes
+    end
+
+    def validate_service_data(data)
+      assert_id_not_specified(data, 'service')
+
+      @collection_klasses = {:ext_management_systems  => ExtManagementSystem,
```

Comment:
The whole `@collection_klasses` thing seems overly complicated - is there any reason not to use the class directly in the `resource_search` calls below? Does the indirection save us anything?